edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -614,29 +614,27 @@ public Object doInHibernate(final Session session) throws HibernateException {
 
 	}
 
+
 	@Override
 	public Long addAssignment(final String gradebookUid, final org.sakaiproject.service.gradebook.shared.Assignment assignmentDefinition) {
 		if (!getAuthz().isUserAbleToEditAssessments(gradebookUid)) {
 			log.error("AUTHORIZATION FAILURE: User {} in gradebook {} attempted to add an assignment", getUserUid(), gradebookUid);
 			throw new GradebookSecurityException();
 		}
 
-		// Ensure that points is > zero.
-		final Double points = assignmentDefinition.getPoints();
-		if ((points == null) || (points <= 0)) {
-			throw new AssignmentHasIllegalPointsException("Points must be > 0");
-		}
+		GradebookHelper.validateAssignmentNameAndPoints(assignmentDefinition);
 
 		final Gradebook gradebook = getGradebook(gradebookUid);
+		final String validatedName = StringUtils.trimToNull(assignmentDefinition.getName());
```

Comment:
can you return the validatedName from your helper function instead? is that cleaner?

Reply:
I rewrote both of these methods to return the validatedName based on this suggestion. Eliminated about 20 lines of code.